### PR TITLE
chore: update code snippets to include revocation check and trusted attester list

### PR DESCRIPTION
**Pull request by ntn-x2**
Despite `verifyCredential` and `verifyPresentation` not actually interacting with the chain, we mention everywhere we do that but none of the snippets actually do that. So, I introduced such a check everywhere, and also, where it made sense, introduced a list of trusted attester URIs to check against the credential issuer.

### CODE REVIEW

File: code_examples/sdk_examples/src/core_features/claiming/05_verify_presentation.ts [hunk 1]
```diff
@@ -2,8 +2,30 @@ import * as Kilt from '@kiltprotocol/sdk-js'
...
+  )
+  const attestation = Kilt.Attestation.fromChain(
+    attestationChain,
+    presentation.rootHash
+  )
+  if (attestation.revoked) {
+    throw "Credential has been revoked and hence it's not valid."
+  }
+  if (!trustedAttesterUris.includes(attestation.owner)) {
+    throw `Credential was issued by ${attestation.owner} which is not in the provided list of trusted attesters: ${trustedAttesterUris}.`
```
Comment (weichweich) [inline code_examples/sdk_examples/src/core_features/claiming/05_verify_presentation.ts]: Can we throw an `new Error` and also the quotes should be single quotes

File: code_examples/sdk_examples/src/core_features/getting_started/06_verify_credential.ts [hunk 2]
```diff
@@ -3,6 +3,19 @@ import * as Kilt from '@kiltprotocol/sdk-js'
...
+    const attestationInfo = await api.query.attestation.attestations(
+      credential.rootHash
+    )
+    const attestation = Kilt.Attestation.fromChain(
+      attestationInfo,
+      credential.rootHash
+    )
+    // Verify that the credential is not revoked. Exception caught by the catch {} block below.
+    if (attestation.revoked) {
+      throw 'The credential has been revoked, hence it is not valid.'
```
Comment (weichweich) [inline code_examples/sdk_examples/src/core_features/getting_started/06_verify_credential.ts]: throw new Error

File: code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts [hunk 3]
```diff
@@ -64,9 +64,13 @@ export async function main({ session, verifierKeys }: Param) {
       attestationChain,
       credential.rootHash
     )
+    if (attestation.revoked) {
+      console.log("Credential has been revoked and hence it's not valid.")
+      return
```
Comment (weichweich) [inline code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts]: It's not your code but maybe we should throw here.

File: code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts [hunk 4]
```diff
@@ -52,7 +52,7 @@ export async function main({ session, verifierKeys }: Param) {
     try {
       await Kilt.Credential.verifyPresentation(credential)
     } catch (e) {
-      console.log('credential was invalid:', e)
+      console.log('Credential was invalid:', e)
       return
```
Comment (weichweich) [inline code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts]: It's not your code but maybe we should throw here.

File: code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts [hunk 5]
```diff
@@ -64,9 +64,13 @@ export async function main({ session, verifierKeys }: Param) {
...
       credential.rootHash
     )
+    if (attestation.revoked) {
+      console.log("Credential has been revoked and hence it's not valid.")
+      return
+    }
     if (isTrustedAttester(attestation.owner)) {
       console.log(
-        'The claim is valid. Claimers email:',
+        "The claim is valid. Claimer's email:",
```
Comment (weichweich) [inline code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts:67]: Single quote `yarn fix`

**Review by weichweich — COMMENTED** [review]
Nice! Style tho

File: code_examples/sdk_examples/src/core_features/claiming/05_verify_presentation.ts (hunk 1 above)
Comment (ntn-x2) [reply code_examples/sdk_examples/src/core_features/claiming/05_verify_presentation.ts]: It's an automatic linter fix, which I guess switches to double quotes when the string contains a `'`.

File: code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts (hunk 5 above)
Comment (ntn-x2) [reply code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts:67]: That's actually what `yarn fix` produced 😄

File: code_examples/sdk_examples/src/core_features/claiming/05_verify_presentation.ts (hunk 1 above)
Comment (weichweich) [reply code_examples/sdk_examples/src/core_features/claiming/05_verify_presentation.ts]: Ah never mind than.

File: code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts (hunk 5 above)
Comment (weichweich) [reply code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts:67]: Good to know! 😁

File: code_examples/sdk_examples/src/core_features/claiming/05_verify_presentation.ts (hunk 1 above)
Comment (ntn-x2) [reply code_examples/sdk_examples/src/core_features/claiming/05_verify_presentation.ts]: https://github.com/KILTprotocol/docs/pull/201/commits/501b348790fd59c790f0e72432f491e6e1b1c2d2

File: code_examples/sdk_examples/src/core_features/getting_started/06_verify_credential.ts (hunk 2 above)
Comment (ntn-x2) [reply code_examples/sdk_examples/src/core_features/getting_started/06_verify_credential.ts]: https://github.com/KILTprotocol/docs/pull/201/commits/4e75aad56095fbfc3cd9e04f9dd681aaf724a1aa

File: code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts (hunk 4 above)
Comment (ntn-x2) [reply code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts]: https://github.com/KILTprotocol/docs/pull/201/commits/d63cff7b0b3a61dfb01914c3262e3ff854216082

File: code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts (hunk 3 above)
Comment (ntn-x2) [reply code_examples/sdk_examples/src/dapp/verifier/verifyCredentialMessage.ts]: https://github.com/KILTprotocol/docs/pull/201/commits/d63cff7b0b3a61dfb01914c3262e3ff854216082

**Review by weichweich — APPROVED** [review]

**Comment by rflechtner** [comment]
awesome, thanks for fixing this. This will luckily become less confusing with the VCs

File: code_examples/sdk_examples/src/core_features/claiming/05_verify_presentation.ts [hunk 6]
```diff
@@ -2,8 +2,30 @@ import * as Kilt from '@kiltprotocol/sdk-js'
...
+  const attestationChain = await api.query.attestation.attestations(
+    presentation.rootHash
+  )
+  const attestation = Kilt.Attestation.fromChain(
+    attestationChain,
+    presentation.rootHash
+  )
+  if (attestation.revoked) {
+    throw new Error("Credential has been revoked and hence it's not valid.")
+  }
```
Comment (rflechtner) [inline code_examples/sdk_examples/src/core_features/claiming/05_verify_presentation.ts:27]: unfortunately that is not all; we also need to check that the attestation matches the credential data (delegation & ctype). We got `verifyAgainstCredential` for that, although I'm seeing this seems broken too.... it doesn't check the delegation node id.

**Pull request merged**